product-service/productById.js

```diff
@@ -0,0 +1,51 @@
+"use strict";
+const products = require("./products");
+
+module.exports.getProductById = async (event) => {
+  try {
+    let id = null; 
```

Comment:
no need to initialize id variable

you can use es6 destructuring - `const { id } = event.pathParameters;`
or simply  `const id = event.pathParameters['id'];`

Reply:
It seems that when there is no pathParameter, the property pathParameters does not exist (I had that issue with queryParameters), so this is added to handle such situation and avoid 500 error.